### ISSUE TITLE
[FR] Zettel Open filter title content instead of file name for random named file.

### ISSUE THREAD

**Issue by babygau** [michal-h21/vim-zettel#70]
At the moment, _Zettel Open_ is not helpful for note which is named based on random alphanumeric characters. I think it is more convenient to have an option allow to use _Zettel Open_ which already filtered note title as output. A title could be recognized based on the following

For front-matter style:
```
---
title: loremipsum
...
---
```

or simply markdown header

```
# loremipsum

loremipsum
==========
```

_Zettel Open_ will output content of `loremipsum` only 

My concept could be illustrated by using RipGrep to filtered title content based on front-matter style:

```bash
command! -bang -nargs=* ZettelOpen call fzf#run(fzf#wrap(
      \ {'source': "rg -NIU --pcre2 -- '(?<=---\ntitle: ).*' | cut -d' ' -f2-"}, <bang>0))
```
I'm not very good at Vimscript in general speaking, I hope you could find this request interesting enough. 

The purpose of naming file randomly are great when you don't have to think much about the content of the note and don't have to worry about file naming if you change of mind in future. But this could rise a problem that don't allow us to search note based on file name.

**Comment by michal-h21** [comment]
`ZettelOpen` should show window with contents of actually selected file. It should look like this:

![Snímek z 2020-09-23 14-23-50](https://user-images.githubusercontent.com/504800/94011982-735c8e00-fda8-11ea-89df-66e782ab006b.png)

(these are from my debugging archive, so it is best to not try to understand the contents :)

Do you want to get rid of filenames in the matched lines listing completely?

**Comment by babygau** [comment]
Yes, I'd like to get rid of that encrypted-look file name, no line number, no `title:`, only content of title. The output should be much clearer. For example

```bash
Rg> Lorem |
> Lorem
> Lorem Ipsum
> Lorem Ipsum Lorem
> ....
```

Also with `--prompt` option, we could change `Rg>` to `Open Note:` or something similar.
```bash
Open Note > Lorem |
> Lorem
> Lorem Ipsum
> Lorem Ipsum Lorem
> ....
```

**Comment by michal-h21** [comment]
I've just realized that filenames are actually important in FZF listings, the selected filenames are extracted from here.

**Comment by babygau** [comment]
> I've just realized that filenames are actually important in FZF listings, the selected filenames are extracted from here. 

Thank you for your update. I don't know but I think you could somehow hide the filenames in `FZF`. This screenshot is from [neuron.vim](https://github.com/fiatjaf/neuron.vim)

https://github.com/fiatjaf/neuron.vim/blob/master/screenshot.png

They also use `FZF` for searching notes

**Comment by rattletat** [comment]
You can achieve this by installing ripgrep and putitng this in your vim config:

```vim
let g:zettel_fzf_command = "rg --column --line-number --color=always --smart-case --regexp '^title: (.+)' --replace '$1'"
let g:zettel_fzf_options = ['--delimiter=:', '--with-nth=-1']
```
